be/src/util/buffered_stream.h

```diff
@@ -64,4 +72,41 @@ class BufferedInputStream {
     uint64_t _file_offset = 0;
 };
 
+class SharedBufferedInputStream : public IBufferedInputStream {
+public:
+    struct IORange {
+        int64_t offset;
+        int64_t size;
+        bool operator<(const IORange& x) const { return offset < x.offset; }
+    };
+    struct CoalesceOptions {
+        static constexpr int64_t MB = 1024 * 1024;
+        int64_t max_read_size = 16 * MB;
```

Comment:
merge this pr https://github.com/StarRocks/starrocks/commit/5bf0be63dda73e119decf0cec72b36a669fe54a1

Reply:
OK.